valkyrie/lib/valkyrie/specs/shared_specs/storage_adapter.rb

```diff
@@ -16,10 +16,19 @@ class CustomResource < Valkyrie::Resource
   it { is_expected.to respond_to(:handles?).with_keywords(:id) }
   it { is_expected.to respond_to(:find_by).with_keywords(:id) }
   it { is_expected.to respond_to(:upload).with_keywords(:file, :resource) }
+  it { is_expected.to respond_to(:valid?).with_keywords(:id, :size, :digests) }
 
-  it "can upload and re-fetch a file" do
+  it "can upload, validate, and re-fetch a file" do
     resource = CustomResource.new(id: "test")
+    sha1 = Digest::SHA1.file(file).to_s
+    size = file.size
     expect(uploaded_file = storage_adapter.upload(file: file, resource: resource)).to be_kind_of Valkyrie::StorageAdapter::File
+
+    expect(storage_adapter.checksum(id: uploaded_file.id, digests: [Digest::SHA1.new])).to eq([sha1])
+    expect(storage_adapter.valid?(id: uploaded_file.id, size: size, digests: { sha1: sha1 })).to be_truthy
```

Comment:
Can we make this .to eq false, instead of falsey? Stronger assertion.

Reply:
👍